### PR TITLE
Adding os environment keys to cmd environment

### LINKED ISSUE #1339: kctrl dev fails to install git package
**What steps did you take:**
Downloaded the latest version v0.48.1 of kctrl from the releases and tried to install the [pkg-standalone example](https://github.com/carvel-dev/kapp-controller/tree/develop/examples/packaging-standalone) using `kctrl dev`

**What happened:**
Package installation failed as it could not find `git` in the `$PATH` as below. But, git is available in the PATH.

```
$git version
git version 2.42.0
$ kctrl dev -f packages.yaml -f package install.yaml 
Target cluster 'https://127.0.0.1:56506' (nodes: kind-control-plane)

apiVersion: packaging.carvel.dev/v1alpha1
kind: PackageInstall
metadata:
  creationTimestamp: null
  name: pkg-standalone-demo
  namespace: pkg-standalone
spec:
  packageRef:
    refName: pkg-standalone.test.carvel.dev
    versionSelection:
      constraints: 2.0.0
  serviceAccountName: pkg-standalone-sa
  values:
  - secretRef:
      name: pkg-standalone-demo-values
status:
  conditions: null
  friendlyDescription: ""
  observedGeneration: 0

Reconciling in-memory app/pkg-standalone-demo (namespace: pkg-standalone) ...
==> Executing /usr/local/bin/vendir [vendir sync -f - --lock-file /dev/null]
==> Finished executing /usr/local/bin/vendir


4:30:55PM: Fetch started
4:30:55PM: Fetch failed
	    | vendir: Error: Syncing directory '0':
	    |   Syncing directory '.' with git contents:
	    |     Fetching git repository:
	    |       Git [init]: exec: "git": executable file not found in $PATH (stderr: )
	    |
	    | Fetching resources: Error (see .status.usefulErrorMessage for details)

App tailing error: Timed out waiting for caches to sync

Succeeded
```

**What did you expect:**
Expecting the package to be installed smoothly without any errors.

**Anything else you would like to add:**

**Environment:**

- kapp Controller version (execute `kubectl get deployment -n kapp-controller kapp-controller -o yaml` and the annotation is `kbld.k14s.io/images`):
```
kctrl version 0.48.0
```
- Kubernetes version (use `kubectl version`)
```
$kubectl version
Client Version: v1.28.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.27.3
```

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by rcmadhankumar**
#### What this PR does / why we need it:
kctrl dev fails to install git package. The recent [change](https://github.com/carvel-dev/kapp-controller/commit/50caee3ca668c2d9cb84739174b5c8110cfc4b59) is required for running vendir inside a sidecar. But, the local reconciler running on the local machine, it requires env variables to find git binary in the path.

#### Which issue(s) this PR fixes:

Fixes #1339 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by praveenrewar — COMMENTED** [review]
🚀 
It would be great if we could also add a test for this somehow.

File: cli/test/e2e/dev_test.go [hunk 1]
```diff
@@ -47,8 +48,30 @@ spec:
...
+  fetch:
+  - git:
+      url: https://github.com/k14s/k8s-simple-app-example
+      ref: origin/develop
+      subPath: config-step-2-template
+  template:
+  - ytt: {}
+  deploy:
+  - kapp:
+      inspect: {}
```
Comment (rohitagg2020) [inline cli/test/e2e/dev_test.go]: Not Needed

File: cli/test/e2e/dev_test.go [hunk 2]
```diff
@@ -79,4 +102,105 @@ spec:
 		}
 		require.Exactly(t, expectedOutputRows, replaceAgeAndSinceDeployed(output.Tables[0].Rows))
 	})
+
+	logger.Section("dev deploy git app", func() {
+		out, err := kappCtrl.RunWithOpts([]string{"dev", "-f", "-"}, RunOpts{StdinReader: strings.NewReader(gitAppYaml)})
+		fmt.Printf("\n\n Out: %s \n err: %+v ", out, err)
+	})
+
+	logger.Section("inspect gitApp app resources", func() {
```
Comment (rohitagg2020) [inline cli/test/e2e/dev_test.go:110]: Using kctrl, we can get the app status and ensure that it is reconciled successfully. I think kapp inspect is bit too much.

Comment (rcmadhankumar) [reply cli/test/e2e/dev_test.go:110]: Here we are using local reconciler, so we cannot get the app status using kctrl as there is no real App CR created in the cluster.

File: cli/test/e2e/dev_test.go (hunk 1 above)
Comment (rcmadhankumar) [reply cli/test/e2e/dev_test.go]: Inspect removed.

File: cli/test/e2e/dev_test.go (hunk 2 above)
Comment (praveenrewar) [reply cli/test/e2e/dev_test.go:110]: I think we can just check if one of the resources from the app is present or not using kubectl.

Comment (rcmadhankumar) [reply cli/test/e2e/dev_test.go:110]: Or else shall we use kapp as it is and reduce the number of expected rows only with `Deployment` and `Service` kind objects?
(existing test case for kctrl dev already uses kapp to verify the resources, this test case developed as per the [existing one](https://github.com/carvel-dev/kapp-controller/blob/59d4642bc93ec5d02bcf56c6baa229597b7c2c0e/cli/test/e2e/dev_test.go#L65))

**Review by praveenrewar — APPROVED** [review]

**Pull request merged**